src/main/java/com/marianbastiurea/lifeofbees/Hive.java

```diff
@@ -400,20 +400,37 @@ public void beesDie(Date currentDate) {
     }
 
     public boolean checkIfAll6EggsFrameAre80PercentFull() {
-        int eggsFrameFull = 0;
-        boolean allEggsFrameAreFull = false;
-        EggFrame eggFrame = this.eggFrames.get(eggsFrameFull);
-        if (eggFrames.size() == 6) {
-            for (int i = 1; i < 7; i++) {
-                if (eggFrame.getNumberOfEggs() > 0.8 * eggFrame.getMaxEggPerFrame()) {
-                    eggsFrameFull += 1;
-                }
-            }
+        if (eggFrames.size() != 6) {
+            return false;
         }
-        if (eggsFrameFull == 6) {
-            allEggsFrameAreFull = true;
+
+        for (EggFrame eggFrame : eggFrames) {
+            if (eggFrame.getNumberOfEggs() <= 0.8 * eggFrame.getMaxEggPerFrame()) {
```

Comment:
Move `eggFrame.getNumberOfEggs() <= 0.8 * eggFrame.getMaxEggPerFrame()` logic to new `is80PercentFull` method in `EggFrame` class.